docs/design/2020-09-28-cluster-read-only-mode.md

```diff
@@ -0,0 +1,233 @@
+
+The enabling session should actively poll the system table to check if all instances or sessions have acknowledged the enabling attempt. Once it is confirmed, the `set global read_only = on` will return, and the cluster will be guaranteed that it has entered the read only mode.
+
+If the enabling session is disconnected while waiting for the enabling result, the `read_only` mode will continue to take effect, because the enabling session will only poll the system table after it setted the global variable. And the `read_only` global variable will be `enabling`, not `on`. This behavior is the same when the enabling session is forced to terminate (such as the user hits Ctrl+C to cancel), the user only cancels the polling, and the turning on would not be canceled (unless the user hit the Ctrl+C really fast and the global variable is not written yet, which the value of `read_only` will still be `off`). To make sure the cluster enters `read_only` mode, the user could run `set global read_only = on` in a new session, it will return immediately if the cluster is already `read_only` otherwise it will simply continue polling the SROS table, and block until it learned that all server instances has acknowledged the enabling attempt or a configured time interval has passed, e.g. 10 seconds, explanation see below. **Or the user could actively examine the value of `read_only`, if it turns from `enabling` to `on`, it means the whole cluster is then read only. Note the latter approach will have 2 seconds of latency because of global variable cache. **The user could cancel the enabling or turn the mode back to normal by running `set global read_only = off` at any time.
+
+To allow the `read_only` global variable turn from `enabling` to `on` even there is no poller in the cluster (polling canceled or server downed) , the reporter will also check that if she is the last in the cluster to acknowledge, and set the variable to `on` if it is the case.
+
+If the TiDB cluster is scaling out while the `read_only` mode is enabled, the newly added TiDB server will get the `read_only = on` or `read_only = enabling` value from the global variable system table at start, and it will report this in the SROS system table immediately. Also, the newly added TiDB server will register its joining to PD (once [this PR](https://github.com/pingcap/tidb/pull/17649) gets merged), which could be used to track their report status.
+
+If the TiDB cluster is scaling in or there is server shutdown, or even network split , these servers might never acknowledge their status. In case the poller waits forever, the poller will return with warnings containing the TiDB servers in the cluster that have not yet reported read only to the user after waiting 10 seconds.
+
+
+#### How to identify committing transactions & held locks
+
+As described above, enabling attempts should be blocked when there are committing transactions or held write locks.
```

Comment:
Maybe instead of blocked say "block waiting" to say it is a brief queue and not imply it is a prevention (the behavior of block is ambiguous).